app/assets/javascripts/discourse/app/lib/notification-items/base.js

```diff
@@ -0,0 +1,114 @@
+import { formatUsername, postUrl } from "discourse/lib/utilities";
+import { userPath } from "discourse/lib/url";
+import { emojiUnescape } from "discourse/lib/text";
+import { htmlSafe } from "@ember/template";
+import I18n from "I18n";
+
+export default class NotificationItemBase {
+  constructor(notification, currentUser, siteSettings, site) {
+    this.notification = notification;
+    this.currentUser = currentUser;
+    this.siteSettings = siteSettings;
+    this.site = site;
+  }
+
+  /**
+   * @returns {string[]} An array of addtional classes that should be added to the <li> element of the notification item.
+   */
+  get classNames() {
+    return [];
+  }
+
+  /**
+   * @returns {string} A href/path that the notification item should link to.
+   */
+  get linkHref() {
+    if (this.topicId) {
+      return postUrl(
+        this.notification.slug,
+        this.topicId,
+        this.notification.post_number
+      );
+    }
+    if (this.notification.data.group_id) {
+      return userPath(
+        `${this.notification.data.username}/messages/${this.notification.data.group_name}`
+      );
+    }
+  }
+
+  /**
+   * @returns {string} A title for the notification item. It shows up when the user hovers over the notification item.
+   */
+  get linkTitle() {
+    if (this.notificationName) {
+      return I18n.t(`notifications.titles.${this.notificationName}`);
+    } else {
+      return "";
+    }
+  }
+
+  /**
+   * @returns {string} An icon for the notification item.
+   */
+  get icon() {
+    return `notification.${this.notificationName}`;
+  }
+
+  /**
+   * @returns {string} The label is the first part of the text content displayed in the notification. For example, in a like notification, the username of the user who liked the post is the label. If a falsey value is returned, the label is omitted.
+   */
+  get label() {
+    return this.username;
+  }
+
+  /**
+   * @returns {string} The description is the second part of the text content displayed in the notification. For example, in a like notification, the topic title is the description. If a falsey value is returned, the description is omitted.
+   */
+  get description() {
+    const description = emojiUnescape(this.notification.fancy_title);
+    if (description) {
+      return htmlSafe(description);
+    } else {
+      return this.notification.data.topic_title;
+    }
+  }
+
+  /**
+   * Perform additional things when the notification item is clicked.
```

Comment:
```suggestion
   * Function that is called when the notification item is clicked. <provide an example of how this can be used? triggers a transition to a route perhaps?>. 
```

Reply:
Honestly I can't think of a good use-case for this API. It's not used anywhere in core, however I've added it temporarily for the chat plugin because the chat plugin doesn't provide a `href` for its notifications and relies on JavaScript to transition the user. This is bad for accessibility and we'll move away from this pattern in the chat plugin (see t67611), but for the time being we need this API to be able to use chat and the new menu together.